src/odemis/util/__init__.py

```diff
@@ -389,6 +389,20 @@ def expand_rect(rect, margin):
     return minx - margin, miny - margin, maxx + margin, maxy + margin
 
 
+def get_polygon_bbox(coordinates):
+    """
+    Get the maximum and minimum values for a and b from a list of tuples
+    with shape: [(a1,b1), (a2,b2), ....., (an,bn)]
+
+    :param coordinates: (list of nested tuples (a,b))
```

Comment:
In your implementation coordinates can also be a single tuple, for instance `coordinates = [(0, 2)]`. The output is then `[0, 2, 0, 2]`, I think it would make sense to raise an error when the input is a single coordinate.

Additionally the input tuples can also be more than 2 values long, I think it would also make sense to check for that or mention in the docstring that only the first two values of each tuple are used. For instance you can have `coordinates = [(0, 1, 2), (0, 3)]`, the output is then `(0, 1, 0, 3)`.

Reply:
That functionality has unfortunately been added since python 3.10.